site/app/libraries/routers/WebRouter.php

```diff
@@ -116,10 +110,17 @@ private function loadCourses() {
             $course = $this->parameters['_course'];
             /** @noinspection PhpUnhandledExceptionInspection */
             $this->core->loadConfig($semester, $course);
+            $this->course_loaded = true;
         }
     }
 
     private function loginCheck() {
+        // This is a workaround for backward compatibility
+        // Should be removed after ClassicRouter is killed completely
+        if ($this->core->getConfig()->isCourseLoaded() && !$this->course_loaded) {
```

Comment:
Let's add:
```
if ($this->core->isDebug()) {
    throw new \RuntimeException("Attempted to use router for invalid URL. Please report the sequence of pages/actions you took to get to this exception to API developers.");
}
```

just so internally we keep open a valuable way to force developers to tell us about potential various redirects or URL constructions that we might have missed as they'll get an exception page instead of just a silent redirect.